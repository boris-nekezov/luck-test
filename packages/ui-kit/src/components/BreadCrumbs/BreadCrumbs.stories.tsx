import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BreadCrumbs } from './BreadCrumbs';

export default {
  title: 'BreadCrumbs',
  component: BreadCrumbs,
} as ComponentMeta<typeof BreadCrumbs>;

const Template: ComponentStory<typeof BreadCrumbs> = (args) => <BreadCrumbs {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  steps: [true, false, true, null, null, null, null],
  currentStep: 3
};

Primary.argTypes = {
  steps: {
    description: 'List of values shown whether step was succeeded or not',
  },
  currentStep: {
    type: { name: 'number', required: true },
    description: 'Shows what step user currently at',
    control: {
      type: 'number'
    }
  }
};
