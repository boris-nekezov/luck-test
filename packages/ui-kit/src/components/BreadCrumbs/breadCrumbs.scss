@import '../../styles/app.scss';

.bread-crumbs {
  @include theme-switcher() {
    &__wrapper {
      color: $text-color--light;
    }

    &__step {
      &_wrapper {
        &::before, &::after {
          background-color: $bread-crumbs-connector-color--light;
        }
      }

      &_content {
        background-color: $bread-crumbs-item-background-color--light;
        border-color: $border-color--light;

        &--current::after {
          background-color: $active-primary-color--light;
        }

        &--succeeded {
          border-color: $active-primary-color--light;
          color: $active-primary-color--light;
        }
  
        &--failed {
          border-color: $active-secondary-color--light;
          color: $active-secondary-color--light;
        }
      }
    }
  }

  &__wrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    color: $text-color--dark;
  }

  &__step {
    $step-item-size: 2rem;
    $step-item--current-size: 1rem;
    $connector-height: 0.4rem;
    $connector-border-radius: calc($connector-height / 2);

    &_wrapper {
      display: flex;
      align-items: center;

      &::before, &::after {
        content: '';
        height: $connector-height;
        width: 0.5rem;
        background-color: $bread-crumbs-connector-color--dark;
      }

      &::before {
        margin-right: 0.2rem;
        border-radius: 0 $connector-border-radius $connector-border-radius 0;
      }

      &::after {
        margin-left: 0.2rem;
        border-radius: $connector-border-radius 0 0 $connector-border-radius;
      }
    }

    &_content {
      position: relative;
      background-color: $bread-crumbs-item-background-color--dark;
      border: 2px solid $border-color--light;
      width: $step-item-size;
      height: $step-item-size;
      font-weight: 900;

      display: flex;
      justify-content: center;
      align-items: center;

      &, &--current::after {
        box-sizing: border-box;
        border-radius: 50%;
      }

      &--current::after {
        content: '';
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        width: $step-item--current-size;
        height: $step-item--current-size;
        background-color: $active-primary-color--dark;
      }

      &--succeeded {
        border-color: $active-primary-color--dark;
        color: $active-primary-color--dark;
      }

      &--failed {
        border-color: $active-secondary-color--dark;
        color: $active-secondary-color--dark;
      }

      > svg {
        width: 1rem;
        height: 1rem;
      }
    }
  }
}
