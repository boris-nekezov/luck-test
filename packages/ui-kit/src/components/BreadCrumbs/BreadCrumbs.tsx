import React from 'react';
import cn from 'classnames';

import TickIcon from '../../assets/check-solid.svg';
import CrossIcon from '../../assets/cross-solid.svg';

import './breadCrumbs.scss';

export interface BreadCrumbsProps {
  steps: (boolean | null)[];
  currentStep: number;
}

export const BreadCrumbs = (props: BreadCrumbsProps) => {
  const { steps, currentStep } = props;

  return (
    <div className='bread-crumbs bread-crumbs__wrapper'>
      {
        steps.map((succeeded, idx) => {
          const contentClasses = cn('bread-crumbs__step_content', {
            "bread-crumbs__step_content--current": idx === currentStep,
            "bread-crumbs__step_content--succeeded": succeeded === true,
            "bread-crumbs__step_content--failed": succeeded === false,
          });

          return (
            <div key={idx} className='bread-crumbs__step_wrapper'>
              <div className={contentClasses}>
                {succeeded === true && <TickIcon />}
                {succeeded === false && <CrossIcon />}
              </div>
            </div>
          )
        })
      }
    </div>
  )
}
