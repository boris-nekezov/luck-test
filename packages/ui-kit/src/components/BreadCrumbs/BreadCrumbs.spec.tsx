import React from 'react';
import { render } from '@testing-library/react';
import { BreadCrumbs } from './BreadCrumbs';

describe('Given `BreadCrumbs` component', () => {
  let wrapper: ReturnType<typeof render>;

  const stepsMock = [true, false, true, null, null];
  const currentStepMock = 3;

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render(<BreadCrumbs steps={stepsMock} currentStep={currentStepMock} />)
    });

    it('should match snapshot', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    })
  })
});
