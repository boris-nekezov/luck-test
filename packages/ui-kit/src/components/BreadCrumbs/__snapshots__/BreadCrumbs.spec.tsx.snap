// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given \`BreadCrumbs\` component when it is rendered should match snapshot 1`] = `
<body>
  <div>
    <div
      class="bread-crumbs bread-crumbs__wrapper"
    >
      <div
        class="bread-crumbs__step_wrapper"
      >
        <div
          class="bread-crumbs__step_content bread-crumbs__step_content--succeeded"
        >
          <check-solid.svg />
        </div>
      </div>
      <div
        class="bread-crumbs__step_wrapper"
      >
        <div
          class="bread-crumbs__step_content bread-crumbs__step_content--failed"
        >
          <cross-solid.svg />
        </div>
      </div>
      <div
        class="bread-crumbs__step_wrapper"
      >
        <div
          class="bread-crumbs__step_content bread-crumbs__step_content--succeeded"
        >
          <check-solid.svg />
        </div>
      </div>
      <div
        class="bread-crumbs__step_wrapper"
      >
        <div
          class="bread-crumbs__step_content bread-crumbs__step_content--current"
        />
      </div>
      <div
        class="bread-crumbs__step_wrapper"
      >
        <div
          class="bread-crumbs__step_content"
        />
      </div>
    </div>
  </div>
</body>
`;
